api/src/main/java/com/vimeo/networking2/VimeoRequest.kt

```diff
@@ -11,3 +14,24 @@ interface VimeoRequest {
     fun cancel()
 
 }
+
+/**
+ * No-op API request. Used when an invalid param was given by the client. In this case,
+ * the actual API request is not made. So, there is nothing to cancel.
+ */
+class NoOpVimeoRequest: VimeoRequest {
```

Comment:
If this is made to be an `object` then we don't need to allocate a new one every time it is used.